src/pages/requests/index.tsx

```diff
@@ -182,9 +188,10 @@ export function Requests() {
           sourceWithLayer={
             <Source id="ukr_water_needs" type="geojson" data={geojson}>
               {/* @ts-ignore */}
-              <Layer {...layerStyle} filter={layerFilter} />
+              <Layer {...(showRegions ? layerStyleWithRegions : layerStyle)} filter={layerFilter} />
```

Comment:
Ok, now I saw that we are passing always one layer but it's a combined one when showRegions is enabled. You could ignore my previous comment if it's not the case.